knative-operator/pkg/controller/knativeeventing/knativeeventing_controller_test.go

```diff
@@ -61,65 +63,51 @@ func TestEventingReconcile(t *testing.T) {
-			deleted:        true,
-		},
-		{
-			name:           "reconcile request with different KnativeServing owner name",
-			ownerName:      "FOO",
-			ownerNamespace: "knative-eventing",
-			deleted:        false,
-		},
-		{
-			name:           "reconcile request with different KnativeServing owner namespace",
-			ownerName:      "knative-eventing",
-			ownerNamespace: "FOO",
-			deleted:        false,
-		},
-	}
+	}{{
+		name:           "reconcile request with same KnativeEventing owner",
+		ownerName:      "knative-eventing",
+		ownerNamespace: "knative-eventing",
+		deleted:        true,
+	}, {
+		name:           "reconcile request with different KnativeEventing owner name",
+		ownerName:      "FOO",
+		ownerNamespace: "knative-eventing",
+		deleted:        false,
+	}, {
+		name:           "reconcile request with different KnativeEventing owner namespace",
+		ownerName:      "knative-eventing",
+		ownerNamespace: "FOO",
+		deleted:        false,
+	}}
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			ke := &defaultKnativeEventing
-			ns := &dashboardNamespace
-			monitor := &monitoringv1.ServiceMonitor{}
-			initObjs := []runtime.Object{ke, ns, &eventingNamespace}
-
-			// Register operator types with the runtime scheme.
-			s := scheme.Scheme
-			s.AddKnownTypes(v1alpha1.SchemeGroupVersion, ke)
-			scheme.Scheme.AddKnownTypes(monitoringv1.SchemeGroupVersion, monitor)
-
-			cl := fake.NewFakeClient(initObjs...)
-			r := &ReconcileKnativeEventing{client: cl, scheme: s}
+			cl := fake.NewFakeClient([]runtime.Object{ke, dashboardNamespace, eventingNamespace}...)
+			r := &ReconcileKnativeEventing{client: cl, scheme: scheme.Scheme}
 
 			// Reconcile to initialize
-			if _, err := r.Reconcile(defaultRequest); err != nil {
+			if _, err := r.Reconcile(req); err != nil {
 				t.Fatalf("reconcile: (%v)", err)
 			}
 			// Check if Eventing dashboard configmaps are available
 			brokerCM := &corev1.ConfigMap{}
-			err := cl.Get(context.TODO(), types.NamespacedName{Name: "grafana-dashboard-definition-knative-eventing-broker", Namespace: ns.Name}, brokerCM)
+			err := cl.Get(context.TODO(), types.NamespacedName{Name: "grafana-dashboard-definition-knative-eventing-broker", Namespace: dashboardNamespace.Name}, brokerCM)
 			if err != nil {
 				t.Fatalf("get: (%v)", err)
 			}
 			sourceCM := &corev1.ConfigMap{}
-			err = cl.Get(context.TODO(), types.NamespacedName{Name: "grafana-dashboard-definition-knative-eventing-source", Namespace: ns.Name}, sourceCM)
+			err = cl.Get(context.TODO(), types.NamespacedName{Name: "grafana-dashboard-definition-knative-eventing-source", Namespace: dashboardNamespace.Name}, sourceCM)
 			if err != nil {
 				t.Fatalf("get: (%v)", err)
 			}
 			// Check if the eventing service monitors are installed
 			smFilter := &monitoringv1.ServiceMonitor{}
-			err = cl.Get(context.TODO(), types.NamespacedName{Name: "knative-eventing-metrics-broker-filter", Namespace: ns.Namespace}, smFilter)
```

Comment:
@aliok because this is plain wrong. The `Namespace` field of a namespace is empty. And `ns` is also the wrong namespace.